File: Interviews/LeetCode/Topic-Wise/Strings/Easy/Reverse-Vowels-of-a-String/345.cpp
class Solution {
public:
    string reverseVowels(string s) {
        string vo = "", vowel = "AEIOUaeiou", ans = s;
        for (char c: s) {
            size_t found = vowel.find(c);
            if (found != string::npos) vo += c;
        }
        
        int j = vo.size() - 1;
        for (int i = 0; i < ans.size(); i++) {
            size_t found = vowel.find(ans[i]);
            if (found != string::npos) ans[i] = vo[j--];
        }
        
        return ans;
    }
};

File: Interviews/LeetCode/Topic-Wise/Strings/Easy/Reverse-Vowels-of-a-String/345.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vo, vowel, ans = "", "AEIOUaeiou", list(s)
        for c in s:
            if c in vowel:
                vo += c
        j = len(vo) - 1
        for idx, i in enumerate(ans):
            if i in vowel:
                ans[idx] = vo[j]
                j -= 1

        return "".join(ans)
